services/telemetry/telemetry.go

```diff
@@ -340,26 +340,16 @@ func (s *Service) sendV2RequestWithRetries(ctx context.Context, req *reporter.Re
 func (s *Service) sendV2Request(ctx context.Context, req *reporter.ReportRequest) error {
 	s.l.Debugf("Using %s as telemetry host.", s.v2Host)
 
-	settings, err := models.GetSettings(s.db)
+	ssoDetails, err := models.GetPerconaSSODetails(ctx, s.db.Querier)
 	if err != nil {
 		return err
 	}
 
-	cc, err := saasdial.Dial(ctx, settings.SaaS.SessionID, s.v2Host)
+	_, err = saasdial.Dial(ctx, s.v2Host, ssoDetails.AccessToken.AccessToken)
 	if err != nil {
 		return errors.Wrap(err, "failed to dial")
 	}
-	defer cc.Close() //nolint:errcheck
-
-	if _, err = reporter.NewReporterAPIClient(cc).Report(ctx, req); err != nil {
```

Comment:
Sending a report should not be omitted. Can we extend the Dial with ability to send stuff? Maybe we can even rename `saasdial.Dial` to `saas.DoRequest(ctx, url, accessToken, body)` and use it both for getting and posting? HTTP Method might be another argument or deduced form passing `nil` as body.

Reply:
Added another param into Dial to pass body.